tests/test_type/test_ready_valid.py

```diff
@@ -209,7 +209,6 @@ def test_in_ready_valid():
     assert T.ready.is_input()
     assert T.valid.is_input()
     assert T.data.is_input()
-    T = T.flip()
-    assert T.ready.is_output()
-    assert T.valid.is_output()
-    assert T.data.is_output()
+    with pytest.raises(TypeError) as e:
```

Comment:
should we check the `issubclass` relationship? `assert issubclass(m.In(m.ReadyValid[T]), m.ReadyValid[T]))`

Reply:
Turns out this subclass relationship didn't quite hold for the existing types, had to add some base class logic to resolve it.